storage/gcloud/aio/storage/storage.py

```diff
@@ -115,9 +126,113 @@ def __init__(self, project, service_file, token=None, session=None):
         resp.raise_for_status()
         return await resp.json()
 
-    async def download_as_string(self, bucket, object_name, session=None):
-        return await self.download(bucket, object_name,
-                                   params={'alt': 'media'}, session=session)
+    async def _upload_resumable(self, url: str, bucket: str, object_name: str,
+                                data, headers: dict = None,
+                                session: aiohttp.ClientSession = None, timeout: int = 120,
+                                upload_type_resumable_always: bool = False):
+        # https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload
+        async def initiate_upload_session(object_name: str, headers: dict, session: aiohttp.ClientSession):
```

Comment:
Rather than having functions within functions, could we move these to the same level? ie. `Storage._initiate_upload_session` rather than `Storage._upload_resumable.initiate_upload_session`. This applies to `do_upload` as well.

Reply:
Ah, that does make sense. Mostly I like avoiding inner functions except for edge cases for two reasons:
1. You get variable shadowing cases. For example, here you have `object_name` and others declared both in the inner and outer methods. As a general rule, shadowing variables can lead to a bunch of hard-to-debug edge cases where you accidentally reference the wrong things.
2. (and more importantly in this case) I think these are methods which a power user might want to manually call themselves, ie for advanced resumable session cases and handling things themselves. As a general rule, I like to make sure these sorts of things are available to users with a leading underscore so its obvious that they probably don't want to use them but still _can_ if they really need to.

Reply:
I changed the functions to private methods. I also messed up the url argument so the inner functions were only working by accident. Inner functions are probably better in a procedural code lib.